crates/re_time_panel/src/lib.rs

```diff
@@ -547,12 +544,9 @@ impl TimePanel {
             for component_name in re_data_ui::ui_visible_components(tree.components.keys()) {
                 let data = &tree.components[component_name];
 
-                if !data.times.has_timeline(ctx.rec_cfg.time_ctrl.timeline())
-                    && data.num_timeless_messages() == 0
-                {
-                    continue; // ignore fields that have no data for the current timeline
-                }
-
+                let component_has_data_in_current_timeline =
+                    data.times.has_timeline(ctx.rec_cfg.time_ctrl.timeline())
+                        || data.num_timeless_messages() > 0;
```

Comment:
again, this could be turned into a helper function for even more increased readability (even though this is already an improvement)